waku/factory/waku.nim

```diff
@@ -146,7 +148,32 @@ proc newCircuitRelay(isRelayClient: bool): Relay =
     return RelayClient.new()
   return Relay.new()
 
-proc new*(T: type Waku, confCopy: var WakuNodeConf): Result[Waku, string] =
+proc setupCallbacks(
+    node: WakuNode, conf: WakuNodeConf, callbacks: WakuCallbacks
+): Result[void, string] =
+  if callbacks.isNil():
+    info "No external callbacks to be set"
+    return ok()
+
+  if not callbacks.relayHandler.isNil():
+    if node.wakuRelay.isNil():
+      return err("Cannot configure relayHandler callback without Relay mounted")
+
+    let autoShards = node.getAutoshards(conf.contentTopics).valueOr:
+      return err("Could not get autoshards: " & error)
+
+    let confShards =
+      conf.shards.mapIt(RelayShard(clusterId: conf.clusterId, shardId: uint16(it)))
+    let shards = confShards & autoShards
+
+    for shard in shards:
+      discard node.wakuRelay.subscribe($shard, callbacks.relayHandler)
```

Comment:
Let's avoid using `discard` :)

Reply:
The thing is that the procedure returns the callback that gets registered

https://github.com/waku-org/nwaku/blob/ab0c1d4aa0e545c2c74b7e1ea8045823370eee4d/waku/waku_relay/protocol.nim#L423-L425

There's actually nothing to do with it, it doesn't give us information and we don't need that proc for anything

We also use discard in other places because of this same reason

https://github.com/waku-org/nwaku/blob/ab0c1d4aa0e545c2c74b7e1ea8045823370eee4d/waku/node/waku_node.nim#L241

we could change the return type of wakuRelay's subscribe to void if it isn't used anywhere, but I guess it is out of the scope of this PR

Reply:
ah yes, I absolutely agree. It seems then that we have something not designed properly and we need to address , in a separate one :)